tests/reduction/reduction_with_several_reductions_in_kernel.h

```diff
@@ -0,0 +1,637 @@
+      reducer_3.combine(accessor[nd_item.get_global_id()]);
+      reducer_4.combine(accessor[nd_item.get_global_id()]);
+    };
+  }
+}
+
+/** @brief Construct lambda for interacting with five reducers using sycl::range
+ *         or sycl::nd_range
+ *  @tparam RangeT Type range
+ *  @tparam AccessorT buffer accessor type
+ *  @param accessor Accessor to the buffer
+ *  @retval Lambda that can interact with all reducers
+ */
+template <typename RangeT, typename AccessorT>
+auto get_lambda_for_5_reductions(AccessorT accessor) {
+  if constexpr (std::is_same<RangeT,
+                             decltype(reduction_common::range)>::value) {
+    return [=](sycl::id<1> idx, auto& reducer_1, auto& reducer_2,
+               auto& reducer_3, auto& reducer_4, auto& reducer_5) {
+      reducer_1.combine(accessor[idx]);
+      reducer_2.combine(accessor[idx]);
+      reducer_3.combine(accessor[idx]);
+      reducer_4.combine(accessor[idx]);
+      reducer_5.combine(accessor[idx]);
+    };
+  } else {
+    return [=](sycl::nd_item<1> nd_item, auto& reducer_1, auto& reducer_2,
+               auto& reducer_3, auto& reducer_4, auto& reducer_5) {
+      reducer_1.combine(accessor[nd_item.get_global_id()]);
+      reducer_2.combine(accessor[nd_item.get_global_id()]);
+      reducer_3.combine(accessor[nd_item.get_global_id()]);
+      reducer_4.combine(accessor[nd_item.get_global_id()]);
+      reducer_5.combine(accessor[nd_item.get_global_id()]);
+    };
+  }
+}
+
+/** @brief Construct lambda for interacting with six reducers using sycl::range
+ *         or sycl::nd_range
+ *  @tparam RangeT Type range
+ *  @tparam AccessorT buffer accessor type
+ *  @param accessor Accessor to the buffer
+ *  @retval Lambda that can interact with all reducers
+ */
+template <typename RangeT, typename AccessorT>
+auto get_lambda_for_6_reductions(AccessorT accessor) {
+  if constexpr (std::is_same<RangeT,
+                             decltype(reduction_common::range)>::value) {
+    return
+        [=](sycl::id<1> idx, auto& reducer_1, auto& reducer_2, auto& reducer_3,
+            auto& reducer_4, auto& reducer_5, auto& reducer_6) {
+          reducer_1.combine(accessor[idx]);
+          reducer_2.combine(accessor[idx]);
+          reducer_3.combine(accessor[idx]);
+          reducer_4.combine(accessor[idx]);
+          reducer_5.combine(accessor[idx]);
+          reducer_6.combine(accessor[idx]);
+        };
+  } else {
+    return [=](sycl::nd_item<1> nd_item, auto& reducer_1, auto& reducer_2,
+               auto& reducer_3, auto& reducer_4, auto& reducer_5,
+               auto& reducer_6) {
+      reducer_1.combine(accessor[nd_item.get_global_id()]);
+      reducer_2.combine(accessor[nd_item.get_global_id()]);
+      reducer_3.combine(accessor[nd_item.get_global_id()]);
+      reducer_4.combine(accessor[nd_item.get_global_id()]);
+      reducer_5.combine(accessor[nd_item.get_global_id()]);
+      reducer_6.combine(accessor[nd_item.get_global_id()]);
```

Comment:
You could generalize this and write a single function template taking a reducer pack instead of all these copy paste.

Reply:
I don't have any idea for implementing this function, could You provide more information about this?

Reply:
After thinking to this, this is complex (but concise) meta-programming that will take time to tune while we have already a simple working solution today, even if it relies on copy paste.